nautilus_open_any_terminal/open_any_terminal_extension.py

```diff
@@ -183,28 +192,19 @@ def _checkdecode(s):
 
 def open_terminal_in_file(filename):
     """open the new terminal with correct path"""
-    if filename:
-        # escape filename quotations
-        filename = filename.replace('"', '\\"')
-        if new_tab:
-            call(
-                '{0} {1} {2}"{3}" &'.format(
-                    terminal_cmd,
-                    NEW_TAB_PARAMS[terminal],
-                    TERM_WORKDIR_PARAMS[terminal],
-                    filename,
-                ),
-                shell=True,
-            )
-        else:
-            call(
-                '{0} {1}"{2}" &'.format(
-                    terminal_cmd, TERM_WORKDIR_PARAMS[terminal], filename
-                ),
-                shell=True,
-            )
-    else:
-        call("{0} &".format(terminal_cmd), shell=True)
+    cmd = terminal_cmd.copy()
+    cmd[0] = shutil.which(cmd[0])
```

Comment:
theoretically this could return `None`. and then the whole command string is incorrect. Do we really need this line at all? I guess the system takes care of resolving the executable from the path.